twisted/python/test/test_setup.py

```diff
@@ -7,36 +7,39 @@
 
 
 import os
-import sys
 
 
 from setuptools.dist import Distribution
 
 from twisted.trial.unittest import TestCase
 
-from twisted.python import dist
+from twisted.python import _setup, dist3
 from twisted.python.compat import _PY3
-from twisted.python.dist import (get_setup_args, ConditionalExtension,
-                                 build_scripts_twisted, _EXTRAS_REQUIRE)
-from twisted.python.filepath import FilePath
+from twisted.python._setup import (
+    BuildPy3,
+    get_setup_args,
+    ConditionalExtension,
+    _EXTRAS_REQUIRE,
+    )
 
 
 
 class SetupTests(TestCase):
     """
     Tests for L{get_setup_args}.
     """
+
     def test_conditionalExtensions(self):
         """
-        Passing C{conditionalExtensions} as a list of L{ConditionalExtension}
-        objects to get_setup_args inserts a custom build_ext into the result
-        which knows how to check whether they should be built.
+        Will return the arguments with a custom build_ext which knows how to
+        check whether they should be built.
         """
         good_ext = ConditionalExtension("whatever", ["whatever.c"],
                                         condition=lambda b: True)
         bad_ext = ConditionalExtension("whatever", ["whatever.c"],
                                         condition=lambda b: False)
-        args = get_setup_args(conditionalExtensions=[good_ext, bad_ext])
+        self.patch(_setup, '_EXTENSIONS', [good_ext, bad_ext])
+        args = get_setup_args()
```

Comment:
Per https://github.com/twisted/twisted/pull/467#discussion_r74916242, this would not have a `patch` call...